### PR TITLE
MVP Ncurses interface

### PR DESCRIPTION
**Pull request by WillNilges**
Hi it's me the resident dumb gay lizard again:

Before we merge this:
- [ ] Do some cleanup
- [x]  fetch the actual amt of drink credits
- [x] implement dropping a drink
- [x] Handle server-side validation
- [ ] error handling
- [x] Add a way to back out of the app w/o crashing 
- [x] Better menu navigation

### CODE REVIEW

**Comment by WillNilges** [comment]
Might also want to change it so that instead of using a one-time cached JSON blob, you could call the API with `?machine=littledrink` or whatever to get just the slots of one machine.

**Comment by WillNilges** [comment]
I _think_ it's dropping properly, but I don't see my drink balance changing. Wtf?

**Comment by WillNilges** [comment]
Mary help

**Comment by WillNilges** [comment]
I think I might need to just... make another API call to fetch drink credits after a vend.

File: src/ui/inventory.rs [hunk 1]
```diff
@@ -33,28 +32,235 @@ pub fn build_menu(api: &mut api::API, machine_index: i32) {
...
+
+  mvwprintw(win, 1, 3, format!("{} -> SELECT A DRINK", machine_name).as_str());
+  mvwprintw(win, 2, 2, "==========================");
 
-  let inventory: Vec<Item> = api::API::get_inventory(api, machine_index).unwrap();
+  match inventory {
+    Ok(slots) => {
+      // TODO: Get real amt of credits.
+      let mut credits = api::API::get_credits(api);
+      mvwprintw(win, height - 2, width - 20, format!("Credits: {}", credits.unwrap()).as_str());
```
Comment (Mstrodl) [inline src/ui/inventory.rs:45]: should format based on credit str length

File: src/ui/inventory.rs [hunk 2]
```diff
@@ -33,28 +32,235 @@ pub fn build_menu(api: &mut api::API, machine_index: i32) {
...
+              wattron(win, A_REVERSE());
+          }
+          if slots[n].empty {
+            wattron(win, COLOR_PAIR(1));
+          }
+          if !slots[n].active {
+            wattron(win, A_DIM());
+          }
+          mvwprintw(
+              win, 3 + n as i32, 2,
```
Comment (Mstrodl) [inline src/ui/inventory.rs:128]: Magic numbers?

File: src/ui/inventory.rs [hunk 3]
```diff
@@ -33,28 +32,235 @@ pub fn build_menu(api: &mut api::API, machine_index: i32) {
...
+  // TODO: There's a better way to handle these. You could just
+  // Unwrap them, or do something else.
+  let drinks: &Map<String, Value> = match status.as_object() {
+    Some(drinks) => drinks,
+    None => panic!("Fuck"),
+  };
+
+  let machines: &Vec<Value> = match drinks["machines"].as_array() {
+    Some(machines) => machines,
+    None => panic!("Fuck"),
```
Comment (Mstrodl) [inline src/ui/inventory.rs:208]: So better way to handle these cases I've learned is with like https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or

File: src/ui/inventory.rs (hunk 2 above)
Comment (WillNilges) [reply src/ui/inventory.rs:128]: Magic numbers. This is just what ncurses code usually looks like.

**Pull request merged**